tests/Command/DeprecatedGenerateObjectAclCommandTest.php

```diff
@@ -0,0 +1,332 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Command;
+
+use Doctrine\Persistence\ManagerRegistry;
+use PHPUnit\Framework\TestCase;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Command\GenerateObjectAclCommand;
+use Sonata\AdminBundle\Tests\Fixtures\Entity\Foo;
+use Sonata\AdminBundle\Util\ObjectAclManipulatorInterface;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+use Symfony\Bridge\PhpUnit\ExpectDeprecationTrait;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Output\StreamOutput;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Security\Acl\Domain\UserSecurityIdentity;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ *
+ * @group legacy
+ *
+ * NEXT_MAJOR: Remove this class
+ */
+class DeprecatedGenerateObjectAclCommandTest extends TestCase
+{
+    use ExpectDeprecationTrait;
+
+    /**
+     * @var Container
+     */
+    private $container;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->container = new Container();
+    }
+
+    public function testExecuteWithDeprecatedDoctrineService(): void
+    {
+        $this->expectDeprecation('Passing a third argument to %s() is deprecated since sonata-project/admin-bundle 3.x.');
```

Comment:
Can you please move the `expectDeprecation` calls just before where they are triggered?